### PR TITLE
fix(tests): replace hardcoded dead PIDs with get_dead_pid helper

### LINKED ISSUE #418: Flaky test: agent-shutdown-integration.bats prune tests
## Description

`tests/agent-shutdown-integration.bats` has flaky tests that fail intermittently across different runs with different test failures each time. The failures appear race-condition related — they involve PID file operations that depend on process liveness detection.

## Observed failures

**Run 1** (during PR #416 testing):
```
not ok 40 prune keeps alive PIDs and removes dead ones
# (in test file tests/agent-shutdown-integration.bats, line 219)
#   `[ "$status" -ne 0 ]' failed
```

**Run 2** (same PR, same branch, immediate retry):
```
not ok 39 prune removes all dead PIDs and deletes the file
# (in test file tests/agent-shutdown-integration.bats, line 196)
#   `[ ! -f ".vbw-planning/.agent-pids" ]' failed
```

**Run 3** (also during same PR):
```
not ok 128 agent-health: orphan recovery stays within matching legacy team_name scope
# (in test file tests/agent-health.bats, line 318)
```

## Analysis

The tests involve PID file management (`prune` operations) which depend on `kill -0` to detect process liveness. These are timing-sensitive — a process might exit between the PID file write and the prune operation, or conversely a PID might be reused by the OS between checks. The different test names failing on each run confirms this is non-deterministic.

## Impact

These flaky tests do not block merges since the suite passes on retry, but they introduce noise in CI and local development. Observed while testing PR #416 (fix/411-suggest-next-fix-debug-session).

### PR DESCRIPTION
**Pull request by dpearson2699**
Fixes #418

## What

Eliminates flaky PID-related test failures in `agent-shutdown-integration.bats` and `agent-health.bats` by replacing hardcoded "dead" PIDs (99991-99999) with a reliable `get_dead_pid()` helper, and replacing multi-process `echo | grep` pipelines in `agent-pid-tracker.sh` with bash-native `[[ =~ ]]` regex matching.

## Why

Under parallel BATS execution (12 workers), macOS PIDs cycle quickly and hardcoded PIDs 99991-99999 can collide with live processes. When `kill -0` succeeds on a "dead" PID that's actually alive, prune/orphan-recovery tests fail non-deterministically. This is the same root cause as #414 — multi-process operations and hardcoded assumptions that don't hold under parallel execution.

## How

- **`tests/test_helper.bash`**: Added `get_dead_pid()` helper that spawns `sleep 999 &`, kills it, waits for reaping, and verifies via `kill -0` that the PID is actually dead. Includes input validation, SIGKILL fallback, and PID-reuse safety (returns non-zero instead of killing an unrelated process).
- **`tests/agent-shutdown-integration.bats`**: Replaced all hardcoded PIDs (99991-99999) with `get_dead_pid()` calls across 5 prune tests. All calls guarded with `|| fail`.
- **`tests/agent-health.bats`**: Replaced all hardcoded PIDs (99994-99999) with `get_dead_pid()` calls across 5 orphan-recovery tests. All calls guarded with `|| fail`.
- **`tests/agent-health-integration.bats`**: Replaced hardcoded PID 99999 with `get_dead_pid()` in 1 orphan-recovery test. Guarded with `|| fail`.
- **`scripts/agent-pid-tracker.sh`**: Replaced 4 instances of `echo "$pid" | grep -qE '^[1-9][0-9]*$'` with bash-native `[[ "$pid" =~ ^[1-9][0-9]*$ ]]` — eliminates unnecessary subprocesses.

## Acceptance Criteria Verification

1. ✅ `scripts/agent-pid-tracker.sh` uses `[[ =~ ]]` instead of `echo | grep` for PID validation — all 4 instances replaced (lines 50, 65, 100, 109)
2. ✅ `tests/test_helper.bash` provides `get_dead_pid()` that spawns→kills→waits a process — function at lines 46-60 with input validation, SIGKILL fallback, kill -0 verification, and PID-reuse safety
3. ✅ All hardcoded PIDs 99991-99999 removed from test files — zero occurrences remain
4. ✅ BATS suite passes at full parallelism (12 workers) — 2849 passed, 0 failed
5. ✅ All `get_dead_pid()` call sites guarded with `|| fail "get_dead_pid failed"` — 18 call sites across 3 test files

## Testing

- [x] `bash testing/run-all.sh` — 2849 passed, 0 failed (lint + contracts + BATS)
- [x] ShellCheck clean on all modified files
- [x] CI green — all 14 checks passed (8 BATS shards + serial + lint + contracts + QA evidence + linked issue)

## QA Summary

| Phase | Rounds | Model | Findings |
|-------|--------|-------|----------|
| Primary QA (Phase 3) | 3 | Claude (qa-investigator) | R1: 1 low (fixed in 27840a7); R2-R3: 0 |
| Cross-model QA (Phase 3.5) | 1 | GPT-5.4 (qa-investigator-gpt-54) | 0 |
| Copilot PR Review (Phase 4.5) | 5 | GitHub Copilot | R2: 1 (hardened get_dead_pid, 0e9d3bc); R3: 13 (added || fail guards, caef9ad); R4: 1 (PID-reuse safety, b1ee443); R1,R5: 0 |
| **Total** | **10** | | **16 findings fixed, 0 false positives** |

- CI/CD: All 14 required checks green on 33da8e0
- Post-review main sync: Not needed (0 commits behind)

### CODE REVIEW

File: tests/test_helper.bash [hunk 1]
```diff
@@ -41,6 +41,18 @@ teardown_temp_dir() {
...
 
+# Generate a PID that is guaranteed dead. Spawns a short-lived process,
+# kills it, waits for it to exit, and returns its PID. Avoids hardcoded
+# PIDs that may collide with live processes under parallel BATS execution.
+get_dead_pid() {
+  local p
+  sleep 999 &
+  p=$!
+  kill "$p" 2>/dev/null
+  wait "$p" 2>/dev/null || true
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/test_helper.bash:54]: Addressed in commit 0e9d3bc. Added input validation (`[[ -n "$p" ]]`), SIGKILL fallback if SIGTERM fails, and `kill -0` verification that the PID is actually dead before returning.

Kept `sleep 999` instead of `sleep 1` — a short-lived sleep can exit naturally before `kill` runs, creating a PID-recycling race where the OS reassigns the PID to another process. `sleep 999` guarantees the process is alive when killed, which is the whole point of this helper.

File: tests/agent-shutdown-integration.bats [hunk 2]
```diff
@@ -202,7 +207,10 @@ simulate_session_stop() {
   sleep 30 &
   local alive_pid=$!
 
-  printf "${alive_pid}\n99994\n99995\n" > ".vbw-planning/.agent-pids"
+  local dead1 dead2
+  dead1=$(get_dead_pid)
+  dead2=$(get_dead_pid)
+  printf '%s\n%s\n%s\n' "${alive_pid}" "$dead1" "$dead2" > ".vbw-planning/.agent-pids"
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-shutdown-integration.bats:213]: Addressed in caef9ad. Added `|| fail \"get_dead_pid failed\"` guard to all 18 call sites across agent-shutdown-integration.bats, agent-health.bats, and agent-health-integration.bats. Helper failure now produces an immediate test failure instead of silently creating empty PIDs.

File: tests/test_helper.bash [hunk 3]
```diff
@@ -41,6 +41,24 @@ teardown_temp_dir() {
   unset VBW_AGENT_PID_LOCK_DIR _ORIG_HOME _ORIG_GIT_CONFIG_NOSYSTEM _ORIG_GIT_CONFIG_GLOBAL
 }
 
+# Generate a PID that is guaranteed dead. Spawns a short-lived process,
+# kills it, waits for it to exit, and returns its PID. Avoids hardcoded
+# PIDs that may collide with live processes under parallel BATS execution.
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/test_helper.bash]: Addressed in caef9ad. Updated comment to accurately describe the `sleep 999` behavior: "process intended to stay alive until killed" instead of "short-lived process".

File: tests/agent-shutdown-integration.bats [hunk 4]
```diff
@@ -258,10 +270,14 @@ simulate_session_stop() {
 @test "prune recovers from stale lock left by crashed process" {
   cd "$TEST_TEMP_DIR"
   # Simulate stale lock from a crashed process
+  local stale_lock_pid dead1 dead2
+  stale_lock_pid=$(get_dead_pid)
+  dead1=$(get_dead_pid)
+  dead2=$(get_dead_pid)
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-shutdown-integration.bats]: Addressed in caef9ad — all `get_dead_pid` call sites now have `|| fail "get_dead_pid failed"` guard.

File: tests/agent-shutdown-integration.bats [hunk 5]
```diff
@@ -276,9 +292,12 @@ simulate_session_stop() {
 @test "prune recovers from stale lock directory with no pid file" {
   cd "$TEST_TEMP_DIR"
   # Simulate stale lock dir with no pid file (edge case: lock created but pid write failed)
+  local dead1 dead2
+  dead1=$(get_dead_pid)
+  dead2=$(get_dead_pid)
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-shutdown-integration.bats]: Addressed in caef9ad — all `get_dead_pid` call sites now have `|| fail "get_dead_pid failed"` guard.

File: tests/agent-health.bats [hunk 6]
```diff
@@ -172,9 +174,11 @@ EOF
 }
 EOF
 
-  echo '{"pid":"99998","agent_id":"agent-stop-001","agent_type":"vbw:vbw-dev","name":"dev-01"}' | bash "$SCRIPTS_DIR/agent-health.sh" start >/dev/null
+  local dead_pid
+  dead_pid=$(get_dead_pid)
+  echo "{\"pid\":\"$dead_pid\",\"agent_id\":\"agent-stop-001\",\"agent_type\":\"vbw:vbw-dev\",\"name\":\"dev-01\"}" | bash "$SCRIPTS_DIR/agent-health.sh" start >/dev/null
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-health.bats:179]: Addressed in caef9ad — all `get_dead_pid` call sites now have `|| fail "get_dead_pid failed"` guard.

File: tests/agent-health.bats [hunk 7]
```diff
@@ -201,9 +205,12 @@ EOF
...
 
   echo "{\"pid\":\"$LIVE_PID\",\"agent_id\":\"agent-live\",\"agent_type\":\"vbw-dev\"}" | bash "$SCRIPTS_DIR/agent-health.sh" start >/dev/null
-  echo '{"pid":"99997","agent_id":"agent-dead","agent_type":"vbw-dev"}' | bash "$SCRIPTS_DIR/agent-health.sh" start >/dev/null
 
-  run bash -c "echo '{\"pid\":\"99997\",\"agent_id\":\"agent-dead\",\"agent_type\":\"vbw-dev\"}' | bash '$SCRIPTS_DIR/agent-health.sh' stop | jq -r '.hookSpecificOutput.additionalContext'"
+  local dead_pid
+  dead_pid=$(get_dead_pid)
+  echo "{\"pid\":\"$dead_pid\",\"agent_id\":\"agent-dead\",\"agent_type\":\"vbw-dev\"}" | bash "$SCRIPTS_DIR/agent-health.sh" start >/dev/null
+
+  run bash -c "echo '{\"pid\":\"$dead_pid\",\"agent_id\":\"agent-dead\",\"agent_type\":\"vbw-dev\"}' | bash '$SCRIPTS_DIR/agent-health.sh' stop | jq -r '.hookSpecificOutput.additionalContext'"
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-health.bats:213]: Addressed in caef9ad — all `get_dead_pid` call sites now have `|| fail "get_dead_pid failed"` guard.

File: tests/agent-health.bats [hunk 8]
```diff
@@ -314,7 +321,9 @@ EOF
 
   echo "{\"teammate_name\":\"dev-01\",\"team_name\":\"vbw-map-duo\",\"pid\":\"$LIVE_PID\"}" | bash "$SCRIPTS_DIR/agent-health.sh" idle >/dev/null
 
-  run bash -c "echo '{\"teammate_name\":\"dev-01\",\"team_name\":\"vbw-phase-01\",\"pid\":\"99995\"}' | bash '$SCRIPTS_DIR/agent-health.sh' idle | jq -r '.hookSpecificOutput.additionalContext'"
+  local dead_pid
+  dead_pid=$(get_dead_pid)
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-health.bats]: Addressed in caef9ad — all `get_dead_pid` call sites now have `|| fail "get_dead_pid failed"` guard.

File: tests/agent-health.bats [hunk 9]
```diff
@@ -143,7 +143,9 @@ run_health_via_wrapper() {
 EOF
 
   # Create health file with dead PID
-  echo '{"pid":"99999","agent_type":"vbw-dev"}' | bash "$SCRIPTS_DIR/agent-health.sh" start >/dev/null
+  local dead_pid
+  dead_pid=$(get_dead_pid)
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-health.bats]: Addressed in caef9ad — all `get_dead_pid` call sites now have `|| fail "get_dead_pid failed"` guard.

File: tests/agent-shutdown-integration.bats [hunk 10]
```diff
@@ -233,13 +241,17 @@ simulate_session_stop() {
 @test "prune ignores leftover .agent-pids.tmp from interrupted prune" {
   cd "$TEST_TEMP_DIR"
   # Simulate interrupted prune that left a stale temp file with a dead PID
-  echo "12345" > ".vbw-planning/.agent-pids.tmp"
+  local stale_pid
+  stale_pid=$(get_dead_pid)
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-shutdown-integration.bats]: Addressed in caef9ad — all `get_dead_pid` call sites now have `|| fail "get_dead_pid failed"` guard.

File: tests/agent-shutdown-integration.bats [hunk 11]
```diff
@@ -233,13 +241,17 @@ simulate_session_stop() {
...
+  echo "$stale_pid" > ".vbw-planning/.agent-pids.tmp"
 
   # Put a live PID in the real file
   sleep 30 &
   local alive_pid=$!
 
-  printf "${alive_pid}\n99996\n" > ".vbw-planning/.agent-pids"
+  local dead1
+  dead1=$(get_dead_pid)
+  printf '%s\n%s\n' "${alive_pid}" "$dead1" > ".vbw-planning/.agent-pids"
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-shutdown-integration.bats:254]: Addressed in caef9ad — all `get_dead_pid` call sites now have `|| fail "get_dead_pid failed"` guard.

File: tests/agent-health.bats [hunk 12]
```diff
@@ -352,7 +361,9 @@ EOF
 }
 EOF
 
-  run bash -c "echo '{\"teammate_name\":\"dev-01\",\"team_name\":\"vbw-phase-01\",\"pid\":\"99994\"}' | bash '$SCRIPTS_DIR/agent-health.sh' idle | jq -r '.hookSpecificOutput.additionalContext'"
+  local dead_pid
+  dead_pid=$(get_dead_pid)
+  run bash -c "echo '{\"teammate_name\":\"dev-01\",\"team_name\":\"vbw-phase-01\",\"pid\":\"$dead_pid\"}' | bash '$SCRIPTS_DIR/agent-health.sh' idle | jq -r '.hookSpecificOutput.additionalContext'"
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-health.bats:366]: Addressed in caef9ad — all `get_dead_pid` call sites now have `|| fail "get_dead_pid failed"` guard.

File: tests/agent-shutdown-integration.bats [hunk 13]
```diff
@@ -186,8 +186,13 @@ simulate_session_stop() {
...
   cd "$TEST_TEMP_DIR"
-  # Write fake dead PIDs
-  printf '99991\n99992\n99993\n' > ".vbw-planning/.agent-pids"
+  # Generate guaranteed-dead PIDs instead of hardcoded values that may
+  # collide with live processes under parallel BATS execution
+  local dead1 dead2 dead3
+  dead1=$(get_dead_pid)
+  dead2=$(get_dead_pid)
+  dead3=$(get_dead_pid)
+  printf '%s\n%s\n%s\n' "$dead1" "$dead2" "$dead3" > ".vbw-planning/.agent-pids"
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-shutdown-integration.bats:195]: Addressed in caef9ad — all `get_dead_pid` call sites now have `|| fail "get_dead_pid failed"` guard.

File: tests/agent-health-integration.bats [hunk 14]
```diff
@@ -66,7 +66,9 @@ teardown() {
 EOF
 
   # Create health file with dead PID
-  echo '{"pid":"99999","agent_type":"vbw-dev"}' | bash "$SCRIPTS_DIR/agent-health.sh" start >/dev/null
+  local dead_pid
+  dead_pid=$(get_dead_pid)
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/agent-health-integration.bats]: Addressed in caef9ad — all `get_dead_pid` call sites now have `|| fail "get_dead_pid failed"` guard.

File: tests/test_helper.bash [hunk 15]
```diff
@@ -41,6 +41,25 @@ teardown_temp_dir() {
...
+  local p
+  sleep 999 &
+  p=$!
+  [[ -n "$p" ]] || return 1
+  kill "$p" 2>/dev/null || kill -9 "$p" 2>/dev/null || true
+  wait "$p" 2>/dev/null || true
+  if kill -0 "$p" 2>/dev/null; then
+    kill -9 "$p" 2>/dev/null || true
+    wait "$p" 2>/dev/null || true
+    kill -0 "$p" 2>/dev/null && return 1
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/test_helper.bash]: Good catch. Addressed in b1ee443. After `wait` reaps the child, a successful `kill -0` means the PID was reused by an unrelated process — sending `kill -9` would kill that process. Replaced the SIGKILL+wait+recheck block with a simple `return 1` so the caller (via `|| fail`) retries or reports the failure.
